### ISSUE TITLE
VerifyError 

### ISSUE THREAD

**Issue by bdaum** [Traneptora/jxlatte#35]
Running the decoder under Java 8 (Temurin) I got a VerifyError in JXLCodestreamDecoder#copyToCanvas().
I could fix it by replacing the ternary expressions by if-then-else:

private static void copyToCanvas(ImageBuffer canvas, Point start, Point off, Dimension size, ImageBuffer frameBuffer) {
        Object[] canvasB; //bd resolved ternary expressions to if-else because of VerifyError
		if (canvas.isInt())
			canvasB = canvas.getIntBuffer();
		else
			canvasB = canvas.getFloatBuffer();
        Object[] frameB;
		if (frameBuffer.isInt())
			frameB = frameBuffer.getIntBuffer();
		else
			frameB = frameBuffer.getFloatBuffer();
        for (int y = 0; y < size.height; y++) {
            System.arraycopy(frameB[y + off.y], off.x, canvasB[y + start.y], start.x, size.width);
        }
    }

**Comment by Traneptora** [comment]
Do you have a sample this can be reproduced by?

**Comment by bdaum** [comment]
No, not an isolated example. I imported the Jxlatte source code into an plugin under Eclipse and Java 1.8 (Temurin) to run within an RCP applicaton. The problem occured with copyToCanvas() as static method, as instance method, and inlined into the calling code. 
An Internet research showed that quite a few Verify errors under Java 1.8 were reported in connection with ternary ( = ? : ) expressions. I resolved the two ternary expression into if-then-else and the problem was gone.
My suspicion is that the Java 1.8 compiler does in this case not handle the type casting well. Both int[][] and float[][] must be cast to the same Object[].
